### PR TITLE
Make the laplacian smoother respect block boundaries too

### PR DESCRIPTION
**Pull request by friedmud**
Also update find_boundary_nodes() so it doesn't build a std::vector<bool> that is n_nodes in length

### CODE REVIEW

File: include/mesh/mesh_tools.h [hunk 1]
```diff
@@ -134,12 +134,27 @@ void build_nodes_to_elem_map (const MeshBase & mesh,
...
+/**
+ * Returns a std::set containing Node IDs for all of the boundary nodes
+ */
+std::set<dof_id_type> find_boundary_nodes(const MeshBase & mesh);
+
+/**
+ * Returns a std::set containing Node IDs for all of the block boundary nodes
+ *
+ * A "block boundary node" is a node that is connected to elemenents from 2
+ * or more blockse
```
Comment (roystgnr) [inline include/mesh/mesh_tools.h]: blockse?

File: include/mesh/mesh_tools.h [hunk 2]
```diff
@@ -134,12 +134,27 @@ void build_nodes_to_elem_map (const MeshBase & mesh,
...
  * is on the domain boundary (true)) or not (false).
  */
 void find_boundary_nodes (const MeshBase & mesh,
                           std::vector<bool> & on_boundary);
+#endif
+
+/**
+ * Returns a std::set containing Node IDs for all of the boundary nodes
+ */
+std::set<dof_id_type> find_boundary_nodes(const MeshBase & mesh);
```
Comment (roystgnr) [inline include/mesh/mesh_tools.h]: Should we be using unordered_set for these?  We never expect them to be sorted, but they might be large enough for O(1) lookup to make a difference.

**Comment by roystgnr** [comment]
:+1:, go ahead and merge after those two tweaks.

**Comment by friedmud** [comment]
Made the changes - I'll let testing finish though

**Pull request merged**